engine/src/main/scala/cromwell/engine/backend/jes/Run.scala

```diff
@@ -25,48 +27,60 @@ object Run  {
-      throw new RuntimeException(message)
-    }
+    logger.debug(s"Command line is: $command")
 
-    def runPipeline: String = {
-      val rpargs = new RunPipelineArgs().setProjectId(pipeline.projectId).setServiceAccount(JesServiceAccount)
+    val runtimeInfo = if (jesJobDescriptor.preemptible) PreemptibleJesRuntimeInfo(command, runtimeAttributes) else NonPreemptibleJesRuntimeInfo(command, runtimeAttributes)
+    val pipeline = new model.Pipeline()
+                    .setProjectId(projectId)
+                    .setDocker(runtimeInfo.docker)
+                    .setResources(runtimeInfo.resources)
+                    .setName(workflow.name)
+                    .setInputParameters(jesParameters.collect({ case i: JesInput => i.toGooglePipelineParameter }).toVector.asJava)
+                    .setOutputParameters(jesParameters.collect({ case i: JesFileOutput => i.toGooglePipelineParameter }).toVector.asJava)
 
-      rpargs.setInputs(pipeline.jesParameters.collect({ case i: JesInput => i.name -> i.toGoogleRunParameter }).toMap.asJava)
+    def runPipeline(): String = {
+      val rpargs = new RunPipelineArgs().setProjectId(projectId).setServiceAccount(JesServiceAccount)
+
+      rpargs.setInputs(jesParameters.collect({ case i: JesInput => i.name -> i.toGoogleRunParameter }).toMap.asJava)
       logger.info(s"Inputs:\n${stringifyMap(rpargs.getInputs.asScala.toMap)}")
 
-      rpargs.setOutputs(pipeline.jesParameters.collect({ case i: JesFileOutput => i.name -> i.toGoogleRunParameter }).toMap.asJava)
+      rpargs.setOutputs(jesParameters.collect({ case i: JesFileOutput => i.name -> i.toGoogleRunParameter }).toMap.asJava)
       logger.info(s"Outputs:\n${stringifyMap(rpargs.getOutputs.asScala.toMap)}")
 
-      val rpr = new RunPipelineRequest().setPipelineId(pipeline.pipelineId.get).setPipelineArgs(rpargs)
+      val rpr = new RunPipelineRequest().setEphemeralPipeline(pipeline).setPipelineArgs(rpargs)
 
       val logging = new LoggingOptions()
-      logging.setGcsPath(s"${pipeline.gcsPath}/${JesBackend.jesLogFilename(pipeline.key)}")
+      logging.setGcsPath(s"$gcsPath/${JesBackend.jesLogFilename(key)}")
       rpargs.setLogging(logging)
 
-      val runId = pipeline.genomicsService.pipelines().run(rpr).execute().getName
+      val runId = genomicsInterface.pipelines().run(rpr).execute().getName
       logger.info(s"JES Run ID is $runId")
       runId
     }
 
-    // Only run the pipeline if the pipeline ID is defined.  The pipeline ID not being defined corresponds to a
-    // resumption of a previous run, and runIdForResumption will be defined.  The Run code takes care of polling
-    // in both the newly created and resumed scenarios.
-    val runId = if (pipeline.pipelineId.isDefined) runPipeline else pipeline.runIdForResumption.get
-    new Run(runId, pipeline, logger)
+    // If runIdForResumption is defined use that, otherwise we'll create a new Run with an ephemeral pipeline.
+    // The Run code takes care of polling.
```

Comment:
I realize this comment was refactored from before but I still don't know what it's trying to say.  😦